ui/src/main/java/com/dat3m/ui/result/ReachabilityResult.java

```diff
@@ -36,22 +43,53 @@ public String getVerdict(){
     private void run(){
         if(validate()){
             VerificationTask task = new VerificationTask(program, wmm, program.getArch() != null ? program.getArch() : options.getTarget(), options.getSettings());
-            Context ctx = new Context();
-            Solver solver = ctx.mkSolver();
-            Result result = null;
-            switch(options.getMethod()) {
-            	case INCREMENTAL:
-            		result = runAnalysisIncrementalSolver(solver, ctx, task);
-            		break;
-            	case ASSUME:
-            		result = runAnalysisAssumeSolver(solver, ctx, task);
-            		break;
-            	case TWOSOLVERS:
-                    result = runAnalysis(solver, ctx, task);
-                    break;
+            Result result = Result.UNKNOWN;
+
+            ShutdownManager sdm = ShutdownManager.create();
+        	Thread t = new Thread(() -> {
+    			try {
+    				if(options.getSettings().getSolverTimeout() > 0) {
+    					// Converts timeout from secs to millisecs
+    					Thread.sleep(1000 * options.getSettings().getSolverTimeout());
+    					sdm.requestShutdown("Shutdown Request");
+    				}
+    			} catch (InterruptedException e) {
+    				// Verification ended, nothing to be done.
+    			}});
+
+            try {
+            	t.start();
+                Configuration config = Configuration.builder()
+                		.setOption("solver.z3.usePhantomReferences", "true")
+                		.build();
+				SolverContext ctx = SolverContextFactory.createSolverContext(
```

Comment:
Use ctx with try-with-resource statement to do proper cleanup. Especially when a timeout interrupts the solving, we end up leaking a context